tsl/src/remote/dist_commands.c

```diff
@@ -75,6 +75,59 @@ ts_dist_cmd_collect_responses(List *requests)
 	return results;
 }
 
+/*
+ * Invoke SQL statement (command) on the given data node connections.
+ */
+static DistCmdResult *
+ts_dist_cmd_invoke_on_data_node_connections(const char *sql, List *data_nodes, List *dn_conn_list)
```

Comment:
I think it would make more sense to introduce `Cache *ts_connection_cache_pin()` function in the same way as we have for `ts_hypertable_cache_pin()`, it would just call `ts_cache_pin` underneath. It is implemented exacly to handle cache invalidations